source/fuzz/transformation_add_parameter.cpp

```diff
@@ -24,37 +24,84 @@ TransformationAddParameter::TransformationAddParameter(
     : message_(message) {}
 
 TransformationAddParameter::TransformationAddParameter(
-    uint32_t function_id, uint32_t parameter_fresh_id, uint32_t initializer_id,
+    uint32_t function_id, uint32_t parameter_fresh_id,
+    std::map<uint32_t, uint32_t>&& call_parameter_id,
     uint32_t function_type_fresh_id) {
   message_.set_function_id(function_id);
   message_.set_parameter_fresh_id(parameter_fresh_id);
-  message_.set_initializer_id(initializer_id);
+  *message_.mutable_call_parameter_id() =
+      fuzzerutil::MapToRepeatedUInt32Pair(call_parameter_id);
   message_.set_function_type_fresh_id(function_type_fresh_id);
 }
 
 bool TransformationAddParameter::IsApplicable(
     opt::IRContext* ir_context, const TransformationContext& /*unused*/) const {
-  // Check that function exists
+  // Check that function exists.
   const auto* function =
       fuzzerutil::FindFunction(ir_context, message_.function_id());
   if (!function ||
       fuzzerutil::FunctionIsEntryPoint(ir_context, function->result_id())) {
     return false;
   }
+  std::map<uint32_t, uint32_t> call_parameter_id_map =
+      fuzzerutil::RepeatedUInt32PairToMap(message_.call_parameter_id());
+  uint32_t new_parameter_type_id = 0;
+  // Iterate over all callers.
+  for (auto* instr :
+       fuzzerutil::GetCallers(ir_context, message_.function_id())) {
+    uint32_t caller_id = instr->result_id();
+
+    // If there is no entry for this caller, return false.
+    if (call_parameter_id_map.find(caller_id) == call_parameter_id_map.end()) {
```

Comment:
Alternatively, you can use `call_parameter_id_map.count(caller_id)`. Its advantage is a smaller amount of code but it's not an ideal solution either. In any way, I'm OK with this approach too.